mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapperTest.java

```diff
@@ -0,0 +1,189 @@
+        cassandra.ensureAllTables();
+        uidValidity = 10;
+        mailbox = new SimpleMailbox<>(new MailboxPath("#private", "benwa@linagora.com", "INBOX"), uidValidity);
+        mailbox.setMailboxId(CassandraId.of(UUID.fromString("464765a0-e4e7-11e4-aba4-710c1de3782b")));
+        maxRetry = 100;
+        cassandraAnnotationMapper = new CassandraAnnotationMapper(mailbox, cassandra.getConf(), maxRetry);
+        executor = Executors.newFixedThreadPool(2);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cassandra.clearAllTables();
+        executor.shutdownNow();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void creatingAnnotaionMapperWithNegativeMaxRetryShouldFail() {
+        new CassandraAnnotationMapper(mailbox, cassandra.getConf(), -1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void creatingAnnotationMapperWithNullMaxRetryShouldFail() {
+        new CassandraAnnotationMapper(mailbox, cassandra.getConf(), 0);
+    }
+
+    @Test
+    public void retrieveEmptyAnnotation() throws Exception {
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(SimpleMailboxAnnotation.EMPTY);
+    }
+
+    @Test
+    public void retrieveAnnotation() throws Exception {
+        cassandra.getConf().execute(
+                insertInto(CassandraAnnotationTable.TABLE_NAME)
+                    .value(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid())
+                    .value(CassandraAnnotationTable.KEY, "/private/comment")
+                    .value(CassandraAnnotationTable.VALUE, "My private comment")
+            );
+
+        Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> resultEntries = buildOneItem();
+        MailboxAnnotation expectedResult = new SimpleMailboxAnnotation(resultEntries);
+
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(expectedResult);
+    }
+
+    @Test
+    public void retrieveMultiAnnotations() throws Exception {
+        Session currentSession = cassandra.getConf();
+        currentSession.execute(
+                insertInto(CassandraAnnotationTable.TABLE_NAME)
+                    .value(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid())
+                    .value(CassandraAnnotationTable.KEY, "/private/comment")
+                    .value(CassandraAnnotationTable.VALUE, "My private comment")
+            );
+        currentSession.execute(
+                insertInto(CassandraAnnotationTable.TABLE_NAME)
+                    .value(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid())
+                    .value(CassandraAnnotationTable.KEY, "/shared/comment")
+                    .value(CassandraAnnotationTable.VALUE, "My shared comment")
+            );
+
+        Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> resultEntries = buildTwoItems();
+        MailboxAnnotation expectedResult = new SimpleMailboxAnnotation(resultEntries);
```

Comment:
if you just want to check entries, you don't have to build the full expected result.